bin/test.py

```diff
@@ -0,0 +1,109 @@
+import os
+import subprocess
+
+import pytest
+import testinfra
+import yaml
+
+ASTRO_IMAGE_NAME = os.environ["ASTRO_IMAGE_NAME"]
+ASTRO_IMAGE_TEST_CONFIG_PATH = os.environ["ASTRO_IMAGE_TEST_CONFIG_PATH"]
+
+os.environ["ASTRO_IMAGE_TAG"] = os.environ.get("CIRCLE_SHA1", "latest")
+test_config = {}
+
+# Read the test config
+if os.path.exists(ASTRO_IMAGE_TEST_CONFIG_PATH):
+    with open(ASTRO_IMAGE_TEST_CONFIG_PATH) as file:
+        config = yaml.safe_load(file)
+
+        if config is not None:
+
+            # Reading test config
+            if "tests" in config:
+                test_config = config["tests"]
+
+
+@pytest.fixture(scope="session")
+def docker_host(request):
+    run_command = ["docker-compose", "run", "-d", ASTRO_IMAGE_NAME]
+
+    # run a container
+    docker_id = subprocess.check_output(run_command).decode().strip()
+
+    # return a testinfra connection to the container
+    yield testinfra.get_host("docker://" + docker_id)
+    # cleanup container after test completion
+    subprocess.check_call(["docker", "rm", "-f", docker_id])
+
+
+#################################################
+#                   User Test                   #
+#################################################
+
```

Comment:
Rather than using large comment blocks like this, if we want to group tests into logical sections we should use classes or separate files.

Reply:
Agree! Just removed them. I don't see much need to have class here.